### PR TITLE
[Image banner] Add ambient movement to background

### LINKED ISSUE #2339: [Image banner] Add ambient movement to background
## Summary

Add ambient animation to the Image Banner section, as prototyped in #2229. 

## Details

This animation will be off by default. It should be implemented as a setting in a new dropdown, like so:
 
Screenshot 2023-02-28 at 11 45 53 AM

### PR DESCRIPTION
**Pull request by kjellr**
### PR Summary: 

Adds an option for ambient animation to the Image Banner section's background image. 

https://user-images.githubusercontent.com/1202812/221939239-9930e1d4-f520-420b-93c2-4f5baf8001ca.mov

### Why are these changes introduced?

Closes #2339. 

### What approach did you take?

This implements a new dropdown menu [as shown in the ticket](#2339). When "Ambient movement" is selected, a CSS class is applied to the media elements within the Image Banner, triggering a long-duration CSS animation that slowly moves the image around within its frame. The CSS for this is defined in `base.css` so that we can theoretically reuse it for the Image with Text (#2340) and Slideshow (#2341) sections. 

As this feature is purely aesthetic, it supports the visitor's `prefers-reduced-motion` preference.

### Visual impact on existing themes

None, this is an optional setting that only adds new functionality. 

### Testing steps/scenarios

1. [Visit the demo store](https://os2-demo.myshopify.com/admin/themes/139450515478/editor). 
2. Add and select an Image Banner. 
3. Look for the new "Image behavior" dropdown underneath the new "Animations" header in the sidebar. 
4. Select "Ambient movement" from that dropdown. 
5. Add various images, verify that they appear as expected.
6. Try various breakpoints, verify that they appear as expected.

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [x] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by stufreen — COMMENTED** [review]
Looking good - a couple minor thoughts

**Comment by kimberlyoleiro** [comment]
Looking great so far to me! Played around with focal points and opacity. Not sure what else we could do to try to break it

File: assets/base.css [hunk 1]
```diff
@@ -3011,3 +3011,17 @@ details-disclosure > details {
     outline: transparent solid 1px;
   }
 }
+
+/* Ambient animation */
+
+@media (prefers-reduced-motion: no-preference) {
+  .animate--ambient > img,
+  .animate--ambient > svg {
+    animation: animateAmbient 30s linear infinite;
```
Comment (melissaperreault) [inline assets/base.css:3020]: It was at 40s before, had a feeling it was a little more perceivable without too much in your face. I curious how others feel about it?

Comment (kjellr) [reply assets/base.css:3020]: I keep going back and forth. 😄 I'll leave it at this setting for a minute and let folks weigh on whether they think it's too fast or not.

File: sections/image-banner.liquid [hunk 2]
```diff
@@ -294,7 +304,27 @@
     },
     {
       "type": "header",
-      "content": "t:sections.image-banner.settings.header.content"
+      "content": "t:sections.image-banner.settings.animations.content"
```
Comment (melissaperreault) [inline sections/image-banner.liquid:307]: Mainly wondering if we should include those settings globally instead, to apply at once on the sections we would include. We would revisit the content if this is the case. Let's regroup to align! 🙏

**Comment by melissaperreault** [comment]
Tested and works like charm! Added some point of discussion, but very excited otherwise! 🎉

File: sections/image-banner.liquid [hunk 3]
```diff
@@ -79,6 +84,11 @@
         else
           assign sizes = '100vw'
         endif
+        if section.settings.image_behavior == 'ambient'
+          assign widths = '450, 660, 900, 1320, 1800, 2136, 2400, 3600, 7680'
```
Comment (stufreen) [inline sections/image-banner.liquid:88]: The `sizes` property is probably the one you want to change here instead (or as well) as the `widths` one.  It's the property that determines which width gets picked so should be scaled by 1.2 as well.

Suggestion:
```
{%- liquid
  if section.settings.image_behavior == 'ambient'
    assign half_width = '60vw'
    assign full_width = '120vw'
    assign two_column_sizes = '(min-width: 750px) 60vw, 120vw'
    assign widths = '450, 660, 900, 1320, 1800, 2136, 2400, 3600, 7680'
  else
    assign half_width = '50vw'
    assign full_width = '100vw'
    assign two_column_sizes = '(min-width: 750px) 50vw, 100vw'
    assign widths = '375, 550, 750, 1100, 1500, 1780, 2000, 3000, 3840'
  endif
-%}
```
Then for each image:
```
        if section.settings.image != blank and section.settings.stack_images_on_mobile
          assign sizes = two_column_sizes
        elsif section.settings.image_2 != blank
          assign sizes = half_width
        else
          assign sizes = full_width
        endif
```

**Review by stufreen — APPROVED** [review]
LGTM - one nit-picky suggestion

File: assets/base.css (hunk 1 above)
Comment (metamoni) [reply assets/base.css:3020]: Would using the `.animate--ambient` selector on its own be enough here? Why do we specifically need to target `img` and `svg` elements?

File: sections/image-banner.liquid (hunk 3 above)
Comment (kjellr) [reply sections/image-banner.liquid]: Thanks, that makes sense! I pushed this change in 01dc560.

**Review by metamoni — APPROVED** [review]
The animation looks great and you've addressed all the feedback in this PR, so this looks good to me! Great job 👍

File: sections/image-banner.liquid [hunk 4]
```diff
@@ -40,17 +54,12 @@
         if section.settings.image_2 != blank
           assign image_class = 'banner__media-image-half'
         endif
-        if section.settings.image_2 != blank and section.settings.stack_images_on_mobile
-          assign sizes = '(min-width: 750px) 50vw, 100vw'
+        if section.settings.image != blank and section.settings.stack_images_on_mobile
```
Comment (stufreen) [inline sections/image-banner.liquid:57]: I think this line shouldn't be changed - should be `section.settings.image_2`.

**Review by stufreen — COMMENTED** [review]
LGTM with one change - plus still waiting for translations?

**Review by metamoni — APPROVED** [review]

**Review by LucasLacerdaUX — APPROVED** [review]
- [x] Single image
- [x] Two images
- [x] Image behaviour
  - [x] None
  - [x] Ambient movement
- [x] Multiple breakpoints
  - [x] Desktop
  - [x] Mobile
- [x] Stacked images on mobile
  - [x] On
  - [x] Off

**Comment by kjellr** [comment]
Translations are in, so I just requested another round of reviews. Thanks, everyone!

**Review by stufreen — APPROVED** [review]
Re-approving - Let's make sure to squash and merge this one too to keep the commit log clean

**Review by metamoni — APPROVED** [review]

**Pull request merged**